boot/core/src/boot/core.clj

```diff
@@ -1218,6 +1218,7 @@
   "A file filtering function factory. FIXME: more documenting here."
   [mkpred]
   (fn [criteria files & [negate?]]
+    (assert (seq criteria) "file-filet (called from by-*) requires a list of matching paths, if you want all files use boot.core/ls")
```

Comment:
There is a lil' typo there and I would drop the `if you want ...` because if you call it from `by-ext` is a weird message back...my proposal:

```
boot.core/file-filter requires a list of criteria but null was passed in, make sure your `by-*` calls are passing them.
```